EcoSoapBank/Network Controller/GraphQLController.swift

```diff
@@ -0,0 +1,105 @@
+//
+//  GraphQLController.swift
+//  EcoSoapBank
+//
+//  Created by Christopher Devito on 8/10/20.
+//  Copyright © 2020 Spencer Curtis. All rights reserved.
+//
+
+import Foundation
+
+enum HTTPMethod: String {
+    case post = "POST"
+}
+
+/// Class containing methods for communicating with GraphQL backend
+class GraphQLController {
+
+    // MARK: - Properties
+
+    private let url = URL(string: "http://35.208.9.187:9094/ios-api-1/")!
+    // Setting up the url request
+    private lazy var request: URLRequest = {
+        var request = URLRequest(url: url)
+        request.httpMethod = HTTPMethod.post.rawValue
+        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+        return request
+    }()
+
+    // MARK: - Request methods
+
+    /// Method for GraphQL query requests
+    /// - Parameters:
+    ///   - query: The intended query in string format
+    ///   - session: The URLSession used for the request. By default this is URLSession.shared
+    ///   - completion: Completion handler that passes back a Result of type Profile or Error
+    func queryRequest(query: String,
+                      session: URLSession = URLSession.shared,
+                      completion: @escaping (Result<Profile, Error>) -> Void) {
+        // Add body to query request
+        let body: [String: String] = ["query": query]
+        request.httpBody = try? JSONSerialization.data(withJSONObject: body, options: [])
+
+        session.dataTask(with: request) { data, _, error in
+            if let error = error {
+                NSLog("\(error)")
+                completion(.failure(error))
+                return
+            }
+
+            guard let data = data else {
+                NSLog("Data is nil")
+                return
+            }
+
+            completion(self.decodeJSON(data: data))
+        }.resume()
+    }
+
+    /// Method for GraphQL mutation requests
+    /// - Parameters:
+    ///   - mutationQuery: The indended mutation query in string format
+    ///   - variables: The variables to be passed in the request
+    ///   - session: The URLSession used for the request. By default this is URLSession.shared
+    ///   - completion: Completion handler that passes back a Result of type Prfile or Error
+    func mutationRequest(mutationQuery: String,
+                         variables: [Any] = [],
+                         session: URLSession = URLSession.shared,
+                         completion: @escaping (Result<Profile, Error>) -> Void) {
```

Comment:
I know Michael Redig passes the session into the method in his version of this sort of thing, but I think it might be simpler to make it a property on the class instance and pass it into the initializer, so that every method just uses the same session (meaning one less parameter for these methods). Later we could make a mock session that conforms to a protocol (like `DataLoader` or something), have URLSession conform to that protocol, and replace `URLSession` with that protocol (but still have `URLSession.shared` be the default.

Reply:
Agreed, sounds like a good plan.